lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -320,7 +320,7 @@ sub update {
                             my $job_template_name;
                             my $prio;
                             my $machine_name;
-                            my $settings = $yaml_defaults_for_arch->{settings};
+                            my %settings = %{$yaml_defaults_for_arch->{settings} // {}};
```

Comment:
This should probably use `Storable::dclone`, which is a bit safer and makes the problem more obvious.